ui/components/component-library/checkbox/checkbox.stories.tsx

```diff
@@ -19,186 +24,154 @@ export default {
     label: {
       control: 'text',
     },
+    name: {
+      control: 'text',
+    },
+    id: {
+      control: 'text',
+    },
   },
-  args: {},
 } as Meta<typeof Checkbox>;
 
 const Template: StoryFn<typeof Checkbox> = (args) => {
-  const [isChecked, setIsChecked] = React.useState(false);
-
+  const [{ isChecked }, updateArgs] = useArgs();
   return (
     <Checkbox
       {...args}
-      onChange={() => setIsChecked(!isChecked)}
-      isChecked={isChecked || args.isChecked}
+      onChange={() =>
+        updateArgs({
+          isChecked: !isChecked,
+        })
+      }
+      isChecked={isChecked}
     />
   );
 };
 
 export const DefaultStory = Template.bind({});
 DefaultStory.storyName = 'Default';
 
-export const Label = (args) => {
-  const [isChecked, setIsChecked] = React.useState(false);
-
-  return (
-    <Checkbox
-      {...args}
-      onChange={() => setIsChecked(!isChecked)}
-      isChecked={isChecked || args.isChecked}
-    />
-  );
-};
-
+export const Label = Template.bind({});
 Label.args = {
-  label: 'Checkbox Label',
-};
-
-export const Id = (args) => {
-  return <Checkbox {...args} />;
+  label: 'Checkbox label',
 };
 
+export const Id = Template.bind({});
 Id.args = {
-  label: 'Id Demo',
+  label: 'Id demo',
   id: 'id-demo',
 };
 
-export const IsChecked = (args) => {
-  return <Checkbox {...args} label="isChecked Demo" />;
-};
-
+export const IsChecked = Template.bind({});
 IsChecked.args = {
   isChecked: true,
+  label: 'isChecked demo',
 };
 
-export const IsIndeterminate = () => {
-  const [isTopCheckboxChecked, setTopCheckboxChecked] = React.useState<
-    boolean | 'indeterminate'
-  >('indeterminate');
-  const [checkboxes, setCheckboxes] = React.useState<boolean[]>([
-    false,
-    true,
-    false,
-  ]);
-
-  const handleTopCheckboxChange = () => {
-    if (
-      isTopCheckboxChecked === true ||
-      isTopCheckboxChecked === 'indeterminate'
-    ) {
-      setTopCheckboxChecked(false);
-      setCheckboxes([false, false, false]);
-    } else {
-      setTopCheckboxChecked(true);
-      setCheckboxes([true, true, true]);
-    }
-  };
+export const IsIndeterminate = (args) => {
+  const [checkedItems, setCheckedItems] = React.useState([false, true, false]);
```

Comment:
Was just playing around with how I saw Chakra do it which I think was interesting, it reduces the amount of code slightly for the functions but both work